src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useContext, useState } from 'react';
+import cn from 'classnames';
+import { MainContext } from '../MainContext/MainContext';
+import { createComments } from '../../api/comments';
+import { Errors } from '../../types/Errors';
+
+export const NewCommentForm: React.FC = () => {
+  const {
+    choosedPost,
+    comments,
+    setComments,
+    setError,
+  } = useContext(MainContext);
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
```

Comment:
Use one `state` since it's one instance (comment)
Create a variable outside the Component
```
const initialCommentState = {
  name: '',
  email: '',
  body: '',
}
```

4. Whenever you need to change value of any field you should pass two arguments `(key, value)` and then just change a value of a comment object by key. In this case you will reuse one handler instead of handlers for each input
Example: 

```
const handleCommentChange = (key, value) => {
  setComment(prev => ({...prev, [key]: value}))
}
```

```suggestion
  const [comment, setComment] = useState(initialCommentState);
```